### PR TITLE
feat(docs): add sentinel-wrapped generated sections to docs/README.md

### PR DESCRIPTION
**Pull request by galligan**
## Summary

- Adds \`<!-- BEGIN:GENERATED:PACKAGE_LIST -->\` / \`<!-- END:GENERATED:PACKAGE_LIST -->\` sentinels to \`docs/README.md\`
- Implements \`generatePackageListSection()\` and \`replaceSentinelSection()\` in \`@outfitter/docs\` core
- Adds \`sync-readme\` subcommand to the docs CLI (\`outfitter-docs docs sync-readme\`)
- Wires \`docs:sync:readme\` in root \`package.json\` through the docs CLI to comply with boundary invocation rules
- Full test coverage for sentinel generation, replacement, and the CLI subcommand

## Test plan

- [ ] \`bun run docs:sync:readme\` updates the generated package list in \`docs/README.md\`
- [ ] Running twice with no changes reports "up to date"
- [ ] Sentinel markers are preserved across updates
- [ ] \`bun run check-boundary-invocations\` passes
- [ ] \`bun run test --filter=@outfitter/docs\` passes

Closes OS-290

Closes: OS-290

<h3>Greptile Summary</h3>

This PR adds automated generation of the package list in `docs/README.md` using HTML comment sentinels. The implementation includes:

- Core `replaceSentinelSection()` function for generic sentinel-based content replacement
- `generatePackageListSection()` to discover publishable packages and format them as a markdown table
- CLI subcommand `sync-readme` accessible via `outfitter-docs docs sync-readme`
- Standalone script `sentinel-sync.ts` for direct invocation
- Full test coverage including success paths, error handling, and edge cases
- Proper boundary invocation through `bunx` in root `package.json`

All previous review feedback has been addressed: sentinel validation is now present in both the CLI command and standalone script, success-path tests were added for the CLI command, and fixture cleanup was implemented in the test suite.

<h3>Confidence Score: 5/5</h3>

- This PR is safe to merge with no blocking issues
- All previous review comments have been resolved. The implementation is well-tested with comprehensive coverage for both success and error paths. The code follows established patterns in the codebase, properly validates sentinel markers before proceeding, and includes appropriate error handling. The boundary invocation through `bunx` is correct, and the generated content matches the expected format.
- No files require special attention

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/docs/src/core/sentinel-generator.ts | New module providing sentinel marker replacement and package list generation. Well-tested with comprehensive coverage. |
| packages/docs/src/core/sentinel-sync.ts | Standalone script with proper sentinel validation added per previous feedback. |
| packages/docs/src/command/create-docs-command.ts | Adds sync-readme subcommand with proper validation and error handling. |
| packages/docs/src/__tests__/sentinel-generator.test.ts | Comprehensive test coverage for sentinel replacement and package discovery. Includes fixture cleanup per previous feedback. |
| packages/docs/src/__tests__/sentinel-sync.test.ts | Tests both success and error paths for the standalone sync script. |
| packages/docs/src/__tests__/docs-command.test.ts | Adds tests for sync-readme CLI command success and error paths per previous feedback. |
| docs/README.md | Updated with sentinel markers and auto-generated package list. Content correctly formatted. |
| package.json | Adds docs:sync:readme script and chains it with docs:sync. Uses bunx for proper boundary invocation. |

</details>

</details>

<details><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[User runs bun run docs:sync:readme] --> B[bunx @outfitter/docs docs sync-readme]
    B --> C[create-docs-command.ts sync-readme action]
    C --> D{Read docs/README.md}
    D --> E{Validate sentinel markers exist?}
    E -->|Missing| F[Exit with error code 1]
    E -->|Present| G[generatePackageListSection]
    G --> H[Discover packages in packages/]
    H --> I[Filter out private packages]
    I --> J[Sort alphabetically by name]
    J --> K[Format as markdown table]
    K --> L[replaceSentinelSection]
    L --> M{Content changed?}
    M -->|No| N[Report 'up to date']
    M -->|Yes| O[Write updated README]
    O --> P[Report success]
```
</details>

<sub>Last reviewed commit: b908573</sub>

### CODE REVIEW

**Comment by galligan** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/537?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#537** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/537?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/537?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#536** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/536?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#535** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/535?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#534** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/534?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#533** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/533?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#532** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/532?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#531** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/531?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#530** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/530?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#529** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/529?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#528** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/528?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#527** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/527?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#526** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/526?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#525** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/525?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#524** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/524?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#523** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/523?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/docs/src/__tests__/docs-command.test.ts [hunk 1]
```diff
@@ -164,4 +175,38 @@ describe("createDocsCommand", () => {
...
+      ["node", "docs", "sync-readme", "--cwd", workspaceRoot],
+      { from: "node" }
+    );
+
+    expect(out).toEqual([]);
+    expect(err).toEqual([
+      "docs/README.md is missing PACKAGE_LIST sentinel markers.",
+    ]);
+    expect(process.exitCode).toBe(1);
+  });
```
[the remark this answers is not in the document]
Comment (galligan) [reply packages/docs/src/__tests__/docs-command.test.ts]: Added. `docs-command.test.ts` now includes a success-path `sync-readme` test that verifies sentinel replacement updates `docs/README.md` and emits the expected success output.

File: packages/docs/src/__tests__/sentinel-generator.test.ts [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    import.meta.dir,
+    "__fixtures__",
+    `sentinel-test-${Date.now()}-${Math.random().toString(36).slice(2, 8)}`
+  );
+  for (const [relativePath, content] of Object.entries(files)) {
+    const filePath = join(tmpDir, relativePath);
+    await mkdir(join(filePath, ".."), { recursive: true });
+    await writeFile(filePath, content, "utf8");
+  }
+  return tmpDir;
```
[the remark this answers is not in the document]
Comment (galligan) [reply packages/docs/src/__tests__/sentinel-generator.test.ts]: Implemented. sentinel-generator tests now track created fixture roots and clean them up in afterEach to avoid fixture directory accumulation across runs.
